### PR TITLE
[Compute] vm create --count mode, support vnet and subnet configuration

### PR DESCRIPTION
**Pull request by qwordy**
**Description**

Resolve https://github.com/Azure/azure-cli/issues/17649
Support --vnet-name, --subnet
Example (3) can work in current version of Azure CLI. Example (1) (2) will work on 5/4/2021. We need to be aware of "az vm create" does not support full functionality of vnet and subnet creation. Use "az network vnet" to create vnet and subnet if you'd like to control all configuration.
**Testing Guide**

```
(1)
# Use existing vnet and subnet.
$rg="fy"
az group create -g $rg -l eastus
az network vnet create -g $rg -n vnet --address-prefix 10.0.0.0/16
az network vnet subnet create -g $rg --vnet-name vnet -n subnet1 --address-prefixes 10.0.0.0/24
az network vnet subnet create -g $rg --vnet-name vnet -n subnet2 --address-prefixes 10.0.1.0/24
az vm create -g $rg -n vma --count 5 --image ubuntults --nsg-rule None --vnet-name vnet --subnet subnet1
az vm create -g $rg -n vmb --count 5 --image ubuntults --nsg-rule None --vnet-name vnet --subnet subnet2

(2)
# Use existing vnet and subnet in another resource group
$rg1="fy2"
$rg2="fy3"
az group create -g $rg1 -l eastus
az group create -g $rg2 -l eastus
az network vnet create -g $rg1 -n vnet --address-prefix 10.0.0.0/16 --subnet-name subnet --subnet-prefix 10.0.0.0/24
$subnetId=$(az network vnet subnet show -g $rg1 -n subnet --vnet-name vnet --query [id] -o tsv)
az vm create -g $rg2 -n vm --count 5 --image ubuntults --nsg-rule None --subnet $subnetID

(3)
# Specify address prefix
$rg="fy4"
$count="150"
az group create -g $rg -l eastus
# Will create new vnet and subnet
az vm create -g $rg -n vma --count $count --image ubuntults --nsg-rule None --vnet-address-prefix 10.0.0.0/8 --subnet-address-prefix 10.0.0.0/16
# Will use vnet and subnet created just now
az vm create -g $rg -n vmb --count $count --image ubuntults --nsg-rule None
az vm create -g $rg -n vmc --count $count --image ubuntults --nsg-rule None
```

**History Notes**

[Component Name 1] BREAKING CHANGE: az command a: Make some customer-facing breaking change.
[Component Name 2] az command b: Add some customer-facing feature.

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [ ] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

File: src/azure-cli/azure/cli/command_modules/vm/_validators.py [hunk 1]
```diff
@@ -1781,9 +1782,7 @@ def _validate_count(namespace):
         namespace.public_ip_address_dns_name,
         namespace.storage_account,
         namespace.storage_container_name,
-        namespace.subnet,
```
Comment (houk-ms) [inline src/azure-cli/azure/cli/command_modules/vm/_validators.py:1784]: why we remove it here?

**Review by houk-ms — COMMENTED** [review]
Can you link your PR which supports the `--count` parameter here? I have no context why we need remove the lines in the validation function.

**Comment by yungezz** [comment]
the change itself LGTM, is there any test on --count, to expand coverage of those params?

**Review by yungezz — APPROVED** [review]

File: src/azure-cli/azure/cli/command_modules/vm/_validators.py (hunk 1 above)
Comment (qwordy) [reply src/azure-cli/azure/cli/command_modules/vm/_validators.py:1784]: Because I disabled it. Now I decide to enable it.

Comment (qwordy) [reply src/azure-cli/azure/cli/command_modules/vm/_validators.py:1784]: https://github.com/Azure/azure-cli/pull/17217

**Comment by qwordy** [comment]
> 
> 
> the change itself LGTM, is there any test on --count, to expand coverage of those params?

I should double all vm test with only one difference "--count". I give up.

**Comment by houk-ms** [comment]
> > the change itself LGTM, is there any test on --count, to expand coverage of those params?
> 
> I should double all vm test with only one difference "--count". I give up.

At least `--vnet-name` and `--subnet` parameters should be tested with `--count`.

**Review by houk-ms — APPROVED** [review]

**Pull request merged**